lib/logstash/inputs/http_poller.rb

```diff
@@ -91,77 +73,101 @@ def run(queue)
     end
   end
 
+  private
   def run_once(queue)
     @urls.each do |name, url|
-      @logger.debug? && @logger.debug("Will get url '#{name}' '#{url}")
-      client.async.get(url).
-        on_success {|response| handle_success(queue, name, url, response)}.
-        on_failure {|exception| handle_failure(queue, name, url, exception)}
+      request_async(queue, name, url)
+    end
+
+    # Some exceptions are only returned here! There is no callback,
+    # for example, if there is a bad port number.
+    client.execute!.each_with_index do |resp, i|
+      if resp.is_a?(java.lang.Exception) || resp.is_a?(StandardError)
+        name = @urls.keys[i]
+        url = @urls[name]
+        handle_failure(queue, name, url, resp)
       end
-    client.execute!
+    end
   end
 
   private
-  def handle_success(queue, name, url, response)
-    @codec.decode(response.body) do |decoded|
-      event = LogStash::Event.new
-
-      event["name"] = name
-      event["host"] = @host
-      event["url"] = url
-      event["responseCode"] = response.code
-      event["message"] = decoded.to_hash
-      event["success"] = response.code >= 200 && response.code <= 299
+  def request_async(queue, name, url)
+    @logger.debug? && @logger.debug("Will get url '#{name}' '#{url}")
+    started = Time.now
+    client.async.get(url).
+      on_success {|response| handle_success(queue, name, url, response, Time.now - started)}.
+      on_failure {|exception| handle_failure(queue, name, url, exception)}
```

Comment:
Do you want to pass the duration `Time.now - started` to the on_failure handler also?